docs/usage/topology_aware_routing.md

```diff
@@ -0,0 +1,96 @@
+
+##### kube-apiserver
+
+The `kube-apiserver` Service is topology-aware. It is consumed by the controllers running in the Shoot control plane.
+
+##### gardener-resource-manager
+
+The `gardener-resource-manager` Service that is part of the Shoot control plane is topology-aware. The resource-manager serves webhooks and the Service is consumed by the kube-apiserver for the webhook communication.
+
+##### vpa-webhook
+
+The `vpa-webhook` Service that is part of the Shoot control plane is topology-aware. It is consumed by the kube-apiserver for the webhook communication.
+
+## Topology-aware Services in the Garden cluster
+
+TBD
+
+## How to make a Service topology-aware?
+
+To make a Service topology-aware, the following annotation and label has to be added to the Service:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+    service.kubernetes.io/topology-aware-hints: "auto"
+  labels:
+    endpoint-slice-hints.resources.gardener.cloud/consider: "true"
+```
+
+The `service.kubernetes.io/topology-aware-hints=auto` annotation is needed for kube-proxy. One of the prerequisites on kube-proxy side for using topology-aware routing is the corresponding Service to be annotated with the `service.kubernetes.io/topology-aware-hints=auto`. For more details, see the following [kube-proxy function](https://github.com/kubernetes/kubernetes/blob/b46a3f887ca979b1a5d14fd39cb1af43e7e5d12d/pkg/proxy/topology.go#L140-L186).
+The `endpoint-slice-hints.resources.gardener.cloud/consider=true` label is needed for gardener-resource-manager to prevent the EndpointSlice hints mutating webhook from selecting all EndpointSlice resources but only the ones that are labeled with the consider label.
+
+The Gardener extensions can use this approach to make a Service they deploy topology-aware.
+
+## How to enable the `TopologyAwareRouting` feature gate?
+
+The `TopologyAwareRouting` feature gate can be enabled in the component config of the corresponding component. For gardenlet:
+
+```yaml
+apiVersion: gardenlet.config.gardener.cloud/v1alpha1
+kind: GardenletConfiguration
+# ...
+featureGates:
+  TopologyAwareRouting: true
+```
+
+For the gardener-operator:
+```yaml
+apiVersion: operator.config.gardener.cloud/v1alpha1
+kind: OperatorConfiguration
+# ...
+featureGates:
+  TopologyAwareRouting: true
+```
+
+For K8s < 1.24 clusters, the `TopologyAwareRouting` feature gate requires the Kubernetes `TopologyAwareHints` feature gate to be enabled for kube-apiserver, kube-controller-manager and kube-proxy. This is required because the `TopologyAwareHints` feature gate is disabled by default in K8s < 1.24. When `TopologyAwareHints` is disabled the kube-apiserver does not allow anything to be persisted in the `.endpoints[].hints` field in the EndpointSlice resource, the kube-controller-manager removes the hints and kube-proxy is not using topology-aware routing.
```

Comment:
```suggestion
⚠️ For K8s < 1.24 clusters, the `TopologyAwareRouting` feature gate requires the Kubernetes `TopologyAwareHints` feature gate to be enabled for kube-apiserver, kube-controller-manager and kube-proxy. This is required because the `TopologyAwareHints` feature gate is disabled by default in K8s < 1.24. When `TopologyAwareHints` is disabled the kube-apiserver does not allow anything to be persisted in the `.endpoints[].hints` field in the EndpointSlice resource, the kube-controller-manager removes the hints and kube-proxy is not using topology-aware routing.
```

You can probably also move this section more to the top because it's such an important information.